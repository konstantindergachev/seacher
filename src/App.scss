.container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  max-width: 800px;
  margin: 20px auto 0;
}
.app__title {
  color: #008080;
}
