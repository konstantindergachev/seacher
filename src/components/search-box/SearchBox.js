import React from 'react';
import './SearchBox.scss';

const SearchBox = ({ placeholder, handleChange }) => {
  return (
    <input
      type="search"
      className="search"
      placeholder={placeholder}
      onChange={handleChange}
    />
  );
};

export default SearchBox;
