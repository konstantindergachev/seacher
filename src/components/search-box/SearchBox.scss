.search {
  border: 1px solid grey;
  border-radius: 5px;
  outline: transparent;
  line-height: 30px;
  // text-transform: uppercase;
  padding: 5px 10px;
  margin: 30px 0;
  width: 50%;
}
