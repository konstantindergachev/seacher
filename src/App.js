import React, { Component } from 'react';
import './App.scss';
import CardList from './components/card-list/CardList';
import SearchBox from './components/search-box/SearchBox';
import Footer from './layout/footer/Footer';

class App extends Component {
  state = {
    data: [],
    searchField: '',
  };

  componentDidMount() {
    this.setState({
      data: this.props.data,
    });
  }

  handleChange = (ev) => {
    this.setState({
      searchField: ev.target.value,
    });
  };

  filterData = (searchFilter) => {
    return this.state.data.filter((item) => {
      return item.name.toLowerCase().includes(searchFilter);
    });
  };

  render() {
    return (
      <div className="container">
        <h1 className="app__title">Monsters Rolodex</h1>
        <SearchBox placeholder="search name" handleChange={this.handleChange} />
        <CardList cards={this.filterData(this.state.searchField)} />
        <Footer />
      </div>
    );
  }
}

export default App;
