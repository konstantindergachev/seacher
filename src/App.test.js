import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';
import SearchBox from './components/search-box/SearchBox';

describe('App component', () => {
  const initialState = {
    data: [
      {
        id: '1',
        name: 'John Snow',
        username: 'JohnJohn',
        email: 'john@john.com',
      },
      {
        id: '2',
        name: 'Tom Magnum',
        username: 'Tomas',
        email: 'tom@tom.com',
      },
    ],
    searchField: '',
  };
  const app = shallow(<App {...initialState} />);
  const tree = renderer.create(<App {...initialState} />).toJSON();
  it('get the snapshot of App component', () => {
    expect(tree).toMatchSnapshot();
  });
  it('render App with SearchBox and Footer', () => {
    expect(app.find('SearchBox')).toHaveLength(1);
    expect(app.find('Footer')).toHaveLength(1);
  });
});

describe('App logic', () => {
  it('initialize App with initial state', () => {
    const initialState = {
      data: [],
      searchField: '',
    };
    const app = shallow(<App {...initialState} />);
    expect(app.state()).toEqual(initialState);
  });
  it('mock on search function', () => {
    const onSearchMock = jest.fn();
    const event = {
      target: { value: '88' },
    };
    const mockProps = {
      placeholder: 'search name',
      handleChange: onSearchMock,
    };
    const searchBox = shallow(<SearchBox {...mockProps} />);
    searchBox.find('input').simulate('change', event.target.value);
    expect(onSearchMock).toBeCalledWith('88');
  });

  it('updates search field in state', () => {
    const initialState = {
      data: [],
      searchField: '',
    };
    const app = shallow(<App {...initialState} />);
    //TODO: Подумать над покрытием строки #20
    // expect(app.state().searchField).toEqual(initialState.searchField);
    // const event = {
    //   target: { value: 'r' },
    // };
    // expect(app.instance().handleChange(event)).toEqual(
    //   initialState.searchField
    // );
  });
  it('filters: data found correctly', () => {
    const initialState = {
      data: [
        {
          id: '1',
          name: 'Tom',
          email: 'tom@tom.com',
        },
      ],
      searchField: 't',
    };
    const app = shallow(<App {...initialState} />);

    expect(app.instance().filterData(initialState.searchField)).toEqual([
      {
        id: '1',
        name: 'Tom',
        email: 'tom@tom.com',
      },
    ]);
  });
  it('filters: data not found', () => {
    const initialState = {
      data: [
        {
          id: '3',
          name: 'John',
          email: 'john@gmail.com',
        },
      ],
      searchField: 'x',
    };
    const filteredData = [];
    const app = shallow(<App {...initialState} />);
    expect(app.instance().filterData(initialState.searchField)).toEqual(
      filteredData
    );
  });
});
