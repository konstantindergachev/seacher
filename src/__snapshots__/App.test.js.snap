// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component get the snapshot of App component 1`] = `
<div
  className="container"
>
  <h1
    className="app__title"
  >
    Monsters Rolodex
  </h1>
  <input
    className="search"
    onChange={[Function]}
    placeholder="search name"
    type="search"
  />
  <div
    className="cards"
  >
    <div
      className="card__container"
    >
      <img
        alt="John Snow"
        className="card__img"
      />
      <div>
        <p
          className="card__name"
        >
          John Snow
        </p>
        <p
          className="card__email"
        >
          john@john.com
        </p>
      </div>
    </div>
    <div
      className="card__container"
    >
      <img
        alt="Tom Magnum"
        className="card__img"
      />
      <div>
        <p
          className="card__name"
        >
          Tom Magnum
        </p>
        <p
          className="card__email"
        >
          tom@tom.com
        </p>
      </div>
    </div>
  </div>
  <footer
    className="footer"
  >
    <span
      className="footer__left"
    >
      Â© 2020
    </span>
    <span
      className="footer__right"
    >
      Konstantin Dergachev
    </span>
  </footer>
</div>
`;
